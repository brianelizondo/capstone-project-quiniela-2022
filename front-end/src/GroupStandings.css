.GroupStandings-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 40px;
}
.GroupStandings-card-title{
    color: rgb(23, 97, 160);
    font-weight: 700;
    font-size: 28px;
    text-transform: uppercase;
}
.GroupStandings-card-table th{
    font-size: 12px;
    text-align: center;
}
.GroupStandings-card-table thead tr th:first-child{
    text-align: left;
}

.GroupStandings-card-table tbody{
    text-align: center;
}
.GroupStandings-card-table tbody tr td:first-child{
    text-align: left;
    font-weight: 700;
}

.GroupStandings-card-team:hover{
    cursor: pointer;
    background-color: rgba(0,0,0, .25);
    transition: 300ms ease-in;
}

.GroupStandings-card button{
    color: white;
    background-color: rgb(60, 154, 236) !important;
    font-weight: 400;
    font-size: 12px;
    border: 3px solid DeepSkyBlue;
    text-transform: uppercase;
}
.GroupStandings-card button:hover{
    color: white;
    background-color: DeepSkyBlue !important;
    transition: 300ms ease-in;
}