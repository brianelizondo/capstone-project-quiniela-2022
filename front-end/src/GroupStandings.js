import React from "react";
import { useHistory } from "react-router-dom";
import { Card, Table, Button } from 'react-bootstrap';

function GroupStandings({ group, standings }){
    let history = useHistory();
    function handleClick(link){
        history.push(link);
    }
    
    return (
        <Card>
            <Card.Body>
                <Card.Title>Group { group }</Card.Title>
                <Table striped size="sm">
                    <thead>
                        <tr>
                            <th>Team</th>
                            <th>Played</th>
                            <th>Won</th>
                            <th>Draws</th>
                            <th>Lost</th>
                            <th>For</th>
                            <th>Against</th>
                            <th>Goal Diff</th>
                            <th>Points</th>
                        </tr>
                    </thead>
                    <tbody>
                        { standings.map((team, idx) => (
                        <tr key={idx} onClick={() => handleClick(`/teams/${team.shortName}`)}>
                            <td>{ team.teamName }</td>
                            <td>{ team.gamesPlayed }</td>
                            <td>{ team.gamesWon }</td>
                            <td>{ team.gamesDraws }</td>
                            <td>{ team.gamesLost }</td>
                            <td>{ team.goalsFor }</td>
                            <td>{ team.goalsAgainst }</td>
                            <td>{ team.goalsDiff }</td>
                            <td>{ team.points }</td>
                        </tr>
                        ))}
                    </tbody>
                </Table>
                <Button variant="info" size="sm" onClick={() => handleClick(`/groups/${group}`)}>Group Details</Button>
            </Card.Body>
        </Card>
    );
}

export default GroupStandings;