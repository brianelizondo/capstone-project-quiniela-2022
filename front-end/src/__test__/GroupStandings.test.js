import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import GroupStandings from '../GroupStandings';

const groupsStandings = [
    {
        group: "A",
        teamID: 1,
        teamName: "Qatar",
        shortName: "QAT",
        gamesPlayed: 0,
        gamesWon: 0,
        gamesDraws: 0,
        gamesLost: 0,
        goalsFor: 0,
        goalsAgainst: 0,
        goalsDiff: 0,
        points: 0
    }
];

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'GroupStandings' component without crashing", async () => {
        render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupStandings group={ "A" } standings={ groupsStandings } detailsButton={false} />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Points");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'GroupStandings' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupStandings group={ "A" } standings={ groupsStandings } detailsButton={false} />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Additional Test", () => {
    it("works when you click on a team name", async () => {
        render(
            <MemoryRouter initialEntries={["/groups/A"]}>
                <GroupStandings group={ "A" } standings={ groupsStandings } detailsButton={false} />
            </MemoryRouter>
        );
        expect(screen.getByText("Group A")).toBeInTheDocument();
        
        // expect a test element to go team details after click
        expect(screen.findByTestId(`team-${groupsStandings[0].shortName}`));
    });
});