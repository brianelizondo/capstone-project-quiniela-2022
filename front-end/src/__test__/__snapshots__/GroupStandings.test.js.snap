// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'GroupStandings' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Group A
      </div>
      <table
        class="table table-sm table-striped"
      >
        <thead>
          <tr>
            <th>
              Team
            </th>
            <th>
              Played
            </th>
            <th>
              Won
            </th>
            <th>
              Draws
            </th>
            <th>
              Lost
            </th>
            <th>
              For
            </th>
            <th>
              Against
            </th>
            <th>
              Goal Diff
            </th>
            <th>
              Points
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="team-QAT"
          >
            <td>
              Qatar
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
            <td>
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
